packages/delisp-core/__tests__/reader.ts

```diff
@@ -116,6 +115,64 @@ describe("Reader", () => {
     });
   });
 
+  it("should read modules", () => {
+    expect(
+      readAllFromString(`
+        (define double
+          (lambda (x) (+ x x)))
+        (define eight (double 4))
+        (log eight)
+      `)
+    ).toMatchObject([
+      {
```

Comment:
Also, the example is quite a pain to maintain if we decide to change it. I would propose to do read the 3 expressions with `readFromString` ,  then concatenate them,  read them with `readAllFromString` and check for a match!

Reply:
also good to add a test to with explicit trailing spaces.

Reply:
Actually, this is a bit tricky to do since the `Location` information is not matching, feels a bit hacky to strip those out completely or have assertions for all properties individually. 

How about just simplifying the expectation to `[list, list, list]` (since the details are covered in other test cases already), or do you have other ideas?

Reply:
Oh, true. I didn't think of that.

Well, stripping location is not that bad I think..  I can imagine the function reading 3 lists but incomplete or something?  Anyway, it's better than nothing, so for now that would also work. If we introduce the bug it would be quite easy to catch and we can add the test then :-)